src/UseCases/CreatePayment/CreatePaymentUseCase.php

```diff
@@ -114,6 +120,26 @@ private function createBankTransferPayment( PaymentCreationRequest $request ): B
 		);
 	}
 
+	/**
+	 * @param PaymentCreationRequest $request
+	 * @return DirectDebitPayment
+	 * @throws PaymentCreationException
+	 */
+	private function createDirectDebitPayment( PaymentCreationRequest $request ): DirectDebitPayment {
+		$iban = new Iban( $request->iban );
+		if ( !$this->validateIbanUseCase->ibanIsValid( $iban ) ) {
```

Comment:
Looking at those 2 lines made me think that we could improve the `ValidateIbanUseCase` by giving it a string input instead of an `Iban` input. This would avoid the necessity for error handling both in this use case (`new Iban` will throw an exception on invalid IBAN) and when a controller passes user input to the `ValidateIbanUseCase`. For efficiency reasons the `ValidateIbanUseCase` could return a success object containing the built `Iban` instance.